app/src/main/java/com/mobileproto/hireddit/hireddit/WordToSpeech.java

```diff
@@ -0,0 +1,37 @@
+package com.mobileproto.hireddit.hireddit;
+
+import android.content.Context;
+import android.speech.tts.TextToSpeech;
+
+import java.util.Locale;
+
+/**
+ * TextToSpeech includes a method which will convert text to corresponding audio output
+ * TextToSpeech constructor gets the application context
+ * Speak() takes a String to generate an audio output
+ * TextToSpeech is able to speak out different languages, but for this project, we set it to English
+ */
+public class WordToSpeech {
+    private TextToSpeech m_ttobj;
```

Comment:
So the WordToSpeech() is a separate class, you can create a WordToSpeech object wherever you want in fragments or somewhere else the code I wrote in Main activity is just to show how to use it(yes, I agree we do not need code in Mainactivity, I can get rid of it).

Yes, the speak() and stop() are not in use right now, because they are created when someone want to use the functionality(you create a WordToSpeech object and call its method speak() to speak out, and call stop() in the onDestroy() in MainActivity to get rid of system leak issue).

About the name, I want to name it TextToSpeech, but it will result in name conflict with the actual TextToSpeech class in android. So I have to change it to something else.

One thing I’m concerned about is that we need to call .stop when the app is destroyed. So I’m kind of hesitated to remove that away from the main activity. Because if the WordToSpeech object is called in multiple fragments, we need to refer to the fragments and call multiple stop() in the onDestroy(). It would be much easier for the fragments to share one speaker created in the main activity and we only need to destroy it once.

We can talked about it tonight so that it will make more sense

Yuzhong



From: Lucy Wilcox [mailto:notifications@github.com]
Sent: Sunday, November 08, 2015 1:44 AM
To: greneroom/Hi-Reddit
Cc: Yuzhong Huang
Subject: Re: [Hi-Reddit] Speech (#5)


In app/src/main/java/com/mobileproto/hireddit/hireddit/WordToSpeech.java<https://github.com/greneroom/Hi-Reddit/pull/5#discussion_r44221277>:

> @@ -0,0 +1,37 @@

> +package com.mobileproto.hireddit.hireddit;

> +

> +import android.content.Context;

> +import android.speech.tts.TextToSpeech;

> +

> +import java.util.Locale;

> +

> +/**

> + * TextToSpeech includes a method which will convert text to corresponding audio output

> + * TextToSpeech constructor gets the application context

> + * Speak() takes a String to generate an audio output

> + * TextToSpeech is able to speak out different languages, but for this project, we set it to English

> + */

> +public class WordToSpeech {

> +    private TextToSpeech m_ttobj;

NORA LOVES COMMENTING ON LUCY'S ACCOUNT: why is this named such?
LUCY: I agree, also no underscores

—
Reply to this email directly or view it on GitHub<https://github.com/greneroom/Hi-Reddit/pull/5/files#r44221277>.